bubble_sort.rb

```diff
@@ -0,0 +1,40 @@
+def bubble_sort(arr)
+  arr.each { |y|
+    x = 0
+    while x < arr.length - 1
+      if arr[x] > arr[x + 1]
+        tmp = arr[x + 1]
+        arr[x + 1] = arr[x]
+        arr[x] = tmp
+      end
+      x += 1
+    end
+  }
+end
+
+def bubble_sort_by(arr)
+  arr.each { |y|
+    x = 0
+    while x < arr.length - 1
+      if arr[x].length > arr[x + 1].length
+        tmp = arr[x + 1]
+        arr[x + 1] = arr[x]
+        arr[x] = tmp
+      end
+      x += 1
+    end
+  }
+end
```

Comment:
`#bubble_sort_by` is supposed to give the user the ability to determine the criteria for sorting the elements.

![odin-ruby-enum](https://user-images.githubusercontent.com/22497270/69281346-69965700-0be8-11ea-9f4d-3633585951b7.png)
[Reference](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks)

_NOTE_
Remember to cover for when a block is not given :wink: